### PR TITLE
Report level filters for graph section

### PR DESCRIPTION
**Pull request by rakeshSgr**

### CODE REVIEW

File: src/app/helpers/environmentVariablesHelper.js [hunk 1]
```diff
@@ -11,8 +11,7 @@ let envVariables = {
   DEFAULT_CHANNEL: env.sunbird_default_channel,
 
   // discussion forum 
-  discussion_forum_token: env.discussion_forum_token || 'a4838b88-6a04-4293-a504-245862cad404',
-  discussions_middleware: env.discussions_middleware || 'http://disussionsmw-service:3002/discussion',
+  discussions_middleware: env.discussions_middleware || 'http://discussionsmw-service:3002',
```
Comment (yravinderkumar33) [inline src/app/helpers/environmentVariablesHelper.js:14]: Please do not modify this file unless n until you're not adding any new variable into Portal...
Please revert this file changes

File: src/app/helpers/environmentVariablesHelper.js [hunk 2]
```diff
@@ -81,6 +80,19 @@ let envVariables = {
   KEYCLOAK_ANDROID_CLIENT: {
     clientId: env.sunbird_android_keycloak_client_id || 'android',
   },
+
```
Comment (yravinderkumar33) [inline src/app/helpers/environmentVariablesHelper.js:83]: revert these changes as well

File: src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.spec.ts [hunk 3]
```diff
@@ -113,77 +113,77 @@ describe('DataChartComponent', () => {
...
-        const spy = spyOn(component, 'buildFiltersForm').and.callThrough();
-        component.ngOnInit();
-        expect(spy).toHaveBeenCalled();
-        expect(spy).toHaveBeenCalledTimes(1);
-        expect(component.filtersFormGroup.contains('Medium')).toBe(true);
-        expect(component.filtersFormGroup.contains('Grade')).toBe(true);
-        expect(component.filtersFormGroup.contains('Textbook name')).toBe(true);
-        expect(component.filtersFormGroup.controls).toBeTruthy();
-    });
+    // it('should build filters form from the configuration', () => {
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.spec.ts:116]: if these tests are no longer valid for this component, please remove them instead of commenting.

File: src/app/client/src/app/modules/dashboard/components/report/report.component.html [hunk 4]
```diff
@@ -257,25 +261,42 @@
                         </div>
                       </div>
 
-                      <div class="ui bottom attached segment p-0 b-0 no-bg my-24" suiTabContent="graph">
+                     <div class="report-block ui bottom attached segment p-0 b-0 no-bg my-24" suiTabContent="graph" >
+                      <div class="" *ngIf="currentReport?.reportMetaData?.filters && !hideElements">
+                       <p class="select-filter-label" > Select Filters <small> (This will apply for all the chart sections)</small> </p>
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/report/report.component.html:266]: No hardcoded labels. Please put this label inside the translation file

Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/report/report.component.html:266]: No hard coded labels

File: src/app/client/src/app/modules/dashboard/components/report/report.component.html [hunk 5]
```diff
@@ -19,7 +19,7 @@
       </div>
       <div class="stackable grid m-0 sb-mid-container-min-height dashboard-body-scroll">
         <div class="sb-sideview-tab-container pb-48 p-0 w-100" *ngIf="showComponent">
-          <ng-container *ngIf="(report$ | async) as currentReport else loading">
+          <ng-container *ngIf="currentReport else loading">
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/report/report.component.html:22]: Please keep using async pipe for auto unsubscription purposes. Also, remove the explicit subscription from the .ts file and include your logic inside this observable.

File: src/app/client/src/app/modules/dashboard/components/filter/filter.component.scss [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    .md-drppicker {
+      padding: calculateRem(16px) calculateRem(8px);
+
+      .ranges ul li {
+        margin-bottom: calculateRem(8px);
+      }
+
+      .btn {
+        background-color: var(--primary-400);
+        margin-right: calculateRem(16px);
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/filter/filter.component.scss:145]: margin-right: calculateRem(16px);
 
 This line is repeated twice. Please remove the duplicate one

File: src/app/client/src/app/modules/dashboard/components/filter/filter.component.html [hunk 7]
```diff
@@ -0,0 +1,87 @@
+  <form [formGroup]="filtersFormGroup" class="ui form"
+  [ngClass]="{'hide': hideElements}" *ngIf="showFilters && !hideElements">
+  <div class="d-flex flex-w-wrap reset-filter-section">
+    <div *ngIf="showGraphStats" class="d-flex flex-ai-center">{{resourceService?.frmelmnts?.lbl?.graphStat }}:
+      <sui-checkbox class="toggle ml-8" appTelemetryInteract [telemetryInteractObject]="telemetryInteractObject"
+        [telemetryInteractEdata]="setTelemetryInteractEdata('graph-statistics')"
+        [telemetryInteractCdata]="telemetryCdata" (checkChange)="statsChange($event)"></sui-checkbox>
+    </div>
+    <div class="mr-8" *ngIf="chartType === 'bar' || chartType === 'line'">
+      <!-- <button (click)="addChartSummary()" *ngIf="chartConfig.id && isUserReportAdmin" type="button"
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/filter/filter.component.html:10]: Why is this piece of code commented out. It's for adding a chart summary...

Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/filter/filter.component.html:10]: this "addChartSummary" function handled in data-chart.component.html

File: src/app/helpers/environmentVariablesHelper.js (hunk 1 above)
Comment (rakeshSgr) [reply src/app/helpers/environmentVariablesHelper.js]: these lines are same as release-3.6.0 branch

File: src/app/helpers/environmentVariablesHelper.js (hunk 2 above)
Comment (rakeshSgr) [reply src/app/helpers/environmentVariablesHelper.js]: these lines are same as release-3.6.0 branch

File: src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.spec.ts (hunk 3 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.spec.ts:116]: removed unwanted code

File: src/app/client/src/app/modules/dashboard/components/report/report.component.html (hunk 4 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/report/report.component.html:266]: made changes

File: src/app/client/src/app/modules/dashboard/components/report/report.component.ts [hunk 8]
```diff
@@ -93,6 +106,13 @@ export class ReportComponent implements OnInit {
         );
       })
     );
+
+    this.report$.subscribe(data => {
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/report/report.component.ts:110]: fixed

File: src/app/client/src/app/modules/dashboard/components/report/report.component.html (hunk 4 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/report/report.component.html:266]: fixed

File: src/app/client/src/app/modules/dashboard/components/report/report.component.html (hunk 5 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/report/report.component.html:22]: fixed

File: src/app/client/src/app/modules/dashboard/components/filter/filter.component.scss (hunk 6 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/filter/filter.component.scss:145]: cleared duplicated code
